auth/client-provider/src/main/java/org/wildfly/security/auth/client/provider/DefaultSSLContextSpi.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2021 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.auth.client.provider;
+
+import org.wildfly.client.config.ConfigXMLParseException;
+import org.wildfly.security.auth.client.AuthenticationContext;
+import org.wildfly.security.auth.client.AuthenticationContextConfigurationClient;
+import org.wildfly.security.auth.client.ElytronXmlParser;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLContextSpi;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLServerSocketFactory;
+import javax.net.ssl.SSLSessionContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+import java.security.AccessController;
+import java.security.GeneralSecurityException;
+import java.security.PrivilegedAction;
+import java.security.SecureRandom;
+
+/**
+ * SSLContextSpi that is used by ElytronClientDefaultSSLContextProvider
+ */
+public class DefaultSSLContextSpi extends SSLContextSpi {
+
+    private SSLContext configuredDefaultClientSSLContext;
+
+    public DefaultSSLContextSpi() throws GeneralSecurityException {
+        this(AuthenticationContext.captureCurrent());
+    }
+
+    public DefaultSSLContextSpi(String configPath) throws GeneralSecurityException, ConfigXMLParseException {
+        this(ElytronXmlParser.parseAuthenticationClientConfiguration(configPathUrlToUri(configPath)).create());
```

Comment:
I think we need some validation for `configPath` here. Looking in the wildfly-config-client repo, it looks like they check that `configPath` is not `null` and it's not empty before calling the corresponding `propertyUrlToUri` method.

Reply:
@fjuma Added IllegalArgumentException if the configPath is null or empty. Thanks!